src/tribler/core/utilities/slow_coro_detection.py

```diff
@@ -0,0 +1,136 @@
+import io
+import logging
+import time
+from asyncio import Handle, Task
+from threading import Lock, Thread
+
+from typing import Optional
+
+from tribler.core.utilities.main_thread_stack_tracking import get_main_thread_stack
+
+from tribler.core.utilities.main_thread_stack_tracking import main_stack_tracking_is_activated
+
+# pylint: disable=protected-access
+
+logger = logging.getLogger(__name__)
+
+SLOW_CORO_DURATION_THRESHOLD = 1.0
+WATCHING_THREAD_INTERVAL = 1.0
+
+
+class DebugInfo:
+    def __init__(self):
+        self.handle = None
+        self.start_time = None
+
+
+current = DebugInfo()
+lock = Lock()
+
+
+class SlowCoroWatchingThread(Thread):
+    """
+    A thread that detects and reports slow coroutines.
+    """
+    def run(self):
+        prev_reported_handle = None  # to detect in the loop when we have the second report of the same slow coroutine
```

Comment:
NIT is the same as above (could be a matter of preference). 
```suggestion
        previous_handle = None  # to detect in the loop when we have the second report of the same slow coroutine
```

Reply:
Done